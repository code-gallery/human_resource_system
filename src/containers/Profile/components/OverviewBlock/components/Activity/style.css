.Activity {
  margin-top: 35px;
  word-wrap: break-word;
}

.Activity-title {
  font-family: 'Graphik-Semibold';
  font-size: 17px;
  line-height: 25px;
  margin-bottom: 25px;
  padding: 0 0 0 25px;
  text-transform: uppercase;
  word-wrap: break-word;
  color: #44474c;
  background-image: url(./assets/activity-ico.svg);
  background-repeat: no-repeat;
  background-position: 0 50%;
  background-size: 14px 25px;
}

.Activity li strong {
  font-family: 'Graphik-Semibold';
  font-size: 11px;
  position: relative;
  display: block;
  width: 100%;
  text-align: left;
  text-transform: uppercase;
  color: #9498a1;
}

.Activity li strong em {
  font-style: normal;
  position: relative;
  z-index: 10;
  display: inline-block;
  padding: 0 10px 0 0;
  background-color: #edf4fc;
}

.Activity li strong:before {
  position: absolute;
  top: 50%;
  left: 0;
  display: block;
  width: 100%;
  height: 1px;
  content: '';
  background-color: #9498a1;
}

.Activity li p {
  font-size: 14px;
  margin: 10px 0 10px 50px;
  color: #44474c;
}

.Activity li p span.activity-verified {
  color: #72d471;
}

.Activity li p span.activity-unverified {
  color: #7f8daa;
}

.Activity li p span.activity-pending {
  color: #e1d65a;
}

.Activity li p span.activity-rejected {
  color: #fb7c7c;
}

.Activity.edit-mode {
  pointer-events: none;
  opacity: .5;
}

.Activity .Activity-pending{
  color: #DDD050;
}

.Activity .Activity-unverified {
  color: #B6C3DF;
}

.Activity .Activity-rejected {
  color: #FA7071;
}

.Activity .Activity-verified {
  color: #67CD66;
}
