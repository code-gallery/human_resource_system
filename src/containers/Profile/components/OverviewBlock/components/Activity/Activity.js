import React from 'react'
import PropTypes from 'prop-types'
import ActivityItem from '../ActivityItem'
import './style.css'

const Activity = ({ activities, editMode }) => (
  <div className={`Activity ${editMode ? 'edit-mode' : ''}`}>
    <p className="Activity-title">Activity</p>
    <ul>
      {
        activities && activities.map((activity, idx) => (
          <ActivityItem
            {...activity}
            key={idx}
          />
        ))
      }
      {!activities.length && (
        <p>No activities yet</p>
      )}
    </ul>
  </div>
)

Activity.propTypes = {
  activities: PropTypes.array,
  editMode: PropTypes.bool
}

export default Activity
