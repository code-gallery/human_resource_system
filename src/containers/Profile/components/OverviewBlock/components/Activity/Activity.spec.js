import React from 'react'
import renderer from 'react-test-renderer'
import Activity from './Activity'

describe('containers/Profile/components/OverviewBlock/.../Activity', () => {
  describe('when activities is empty', () => {
    it('renders correctly', () => {
      const props = {
        activities: [],
        editMode: false
      }
      const tree = renderer.create(
        <Activity {...props} />
      ).toJSON()

      expect(tree).toMatchSnapshot()
    })
  })

  describe('when activities is not empty', () => {
    const activities = [
      {
        assertion: 'developer',
        date: '2017-11-17T17:37:09.000Z',
        organisation: 'Applied Blockchain',
        status: 'pending_verification',
        type: 'job'
      },
      {
        assertion: 'MSc',
        date: '2017-11-17T17:37:09.000Z',
        organisation: 'Westminster',
        status: 'verified',
        type: 'education'
      }
    ]

    describe('when editMode is false', () => {
      it('renders correctly', () => {
        const props = {
          activities,
          editMode: false
        }
        const tree = renderer.create(
          <Activity {...props} />
        ).toJSON()

        expect(tree).toMatchSnapshot()
      })
    })

    describe('when editMode is true', () => {
      it('renders correctly', () => {
        const props = {
          activities,
          editMode: true
        }
        const tree = renderer.create(
          <Activity {...props} />
        ).toJSON()

        expect(tree).toMatchSnapshot()
      })
    })
  })
})
