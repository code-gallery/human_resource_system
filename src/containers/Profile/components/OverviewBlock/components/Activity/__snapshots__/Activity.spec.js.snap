// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/Profile/components/OverviewBlock/.../Activity when activities is empty renders correctly 1`] = `
<div
  className="Activity "
>
  <p
    className="Activity-title"
  >
    Activity
  </p>
  <ul>
    <p>
      No activities yet
    </p>
  </ul>
</div>
`;

exports[`containers/Profile/components/OverviewBlock/.../Activity when activities is not empty when editMode is false renders correctly 1`] = `
<div
  className="Activity "
>
  <p
    className="Activity-title"
  >
    Activity
  </p>
  <ul>
    <li>
      <strong>
        <em>
          17th November 17
        </em>
      </strong>
      <p>
        developer
         at 
        Applied Blockchain
         
        is
         
        <span
          className="Activity-pending"
        >
          pending verification
        </span>
        .
      </p>
    </li>
    <li>
      <strong>
        <em>
          17th November 17
        </em>
      </strong>
      <p>
        MSc
         at 
        Westminster
         
        was
         
        <span
          className="Activity-verified"
        >
          verified
        </span>
        .
      </p>
    </li>
  </ul>
</div>
`;

exports[`containers/Profile/components/OverviewBlock/.../Activity when activities is not empty when editMode is true renders correctly 1`] = `
<div
  className="Activity edit-mode"
>
  <p
    className="Activity-title"
  >
    Activity
  </p>
  <ul>
    <li>
      <strong>
        <em>
          17th November 17
        </em>
      </strong>
      <p>
        developer
         at 
        Applied Blockchain
         
        is
         
        <span
          className="Activity-pending"
        >
          pending verification
        </span>
        .
      </p>
    </li>
    <li>
      <strong>
        <em>
          17th November 17
        </em>
      </strong>
      <p>
        MSc
         at 
        Westminster
         
        was
         
        <span
          className="Activity-verified"
        >
          verified
        </span>
        .
      </p>
    </li>
  </ul>
</div>
`;
